addons/explosives/functions/fnc_allowDefuse.sqf

```diff
@@ -0,0 +1,30 @@
+#include "script_component.hpp"
+/*
+ * Author: Walthzer
+ * Excludes a passed mine from the dynamic defuse action.
+ *
+ * Arguments:
+ * 0: Mine <OBJECT>
+ * 1: Allow defusal <BOOL>
+ *
+ * Return Value:
+ * Succes <BOOLEAN>
+ *
+ * Example:
+ * [_mine, false] call ace_explosives_fnc_allowDefuse
+ *
+ * Public: Yes
+ */
+
+params [["_mine", objNull, [objNull]], ["_allow", true, [true]]];
+TRACE_1("params",_mine,_allow);
+
+if !(_mine in allMines) exitWith {false};
+
+if (_allow) then {
+    if ([_mine] call FUNC(isAllowedDefuse)) exitWith {false};
+    GVAR(excludedMines) = GVAR(excludedMines) - [_mine];
+    true
+} else {
+    not ((GVAR(excludedMines) pushBackUnique _mine) == -1);
+};
```

Comment:
What about this?
```suggestion
if (_allow && {!([_mine] call FUNC(isAllowedDefuse))}) exitWith {
    GVAR(excludedMines) = GVAR(excludedMines) - [_mine];
    true
};

if (!_allow) exitWith {
    (GVAR(excludedMines) pushBackUnique _mine != -1
};

false
```

Reply:
That was my first thought, but I thought it was better to reduce the amount of `if` statements instead of increasing the readability?

Reply:
Even then, this won't run periodically, so readability definitely comes first. :)

Reply:
True, taking that with me for next time.